tools/converter/mule2ballerina/src/main/java/org/wso2/ei/tools/mule2ballerina/configreader/ConfigReader.java

```diff
@@ -214,14 +227,37 @@ private void populateIntermediateObject(Iterator<Attribute> attributes, Class<?>
                 }
             });
 
-            BaseObject muleObj = (BaseObject) object;
-            DataCarrierDTO dataCarrierDTO = populateDataCarrier(muleObj);
-            muleObj.buildTree(dataCarrierDTO);
+            BaseObject baseObj = (BaseObject) finalObject;
+            dataCarrierDTO = populateDataCarrier(baseObj);
+            //If it's not a database object add it to tree (postpone adding db object until query is set)
+            if (!(baseObj instanceof Database)) {
+                baseObj.buildTree(dataCarrierDTO);
+            }
 
         } catch (IllegalAccessException e) {
             logger.error(e.getMessage(), e);
         } catch (InstantiationException e) {
             logger.error(e.getMessage(), e);
+        } catch (ClassNotFoundException e) {
+            logger.error(e.getMessage(), e);
```

Comment:
Any reason for not throwing the exception further ?

Reply:
At this point even if this type of an error occurred conversion process should not halt. and proper messages were added in here >> https://github.com/arukshani/product-ei/commit/5bc332461335b908d9a95df22ffdb43bd8ebe7c0#diff-ea4cf52b0ef514be466d6bf957833ba3L226